### PR TITLE
adding hero custom pattern for simple websites

### PR DESCRIPTION
**Pull request by abhijitb**
## Proposed changes
These changes add a custom hero pattern if the site type matches for a simple homepage.
Screenshot 2024-06-30 at 20 39 25
Screenshot 2024-06-30 at 20 41 08

## Type of Change

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/bluehost/.github/blob/master/.github/contributing.md) doc
- [x] Linting and tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: includes/SiteGen/SiteGen.php [hunk 1]
```diff
@@ -438,9 +439,21 @@ public static function get_home_pages( $site_description, $content_style, $targe
...
+		$generated_patterns = self::get_sitegen_from_cache( 'generatedPatterns' );
+
+		// check if custom hero patterns needs to be added
+		$site_classification = self::get_sitegen_from_cache( 'siteclassification' );
+		if ( Patterns::check_custom_hero_needed( $site_classification ) ) {
+			// update content structures and generated patterns
+			$custom_structure = Patterns::get_custom_content_structure();
+			foreach ( $generated_content_structures as $home_slug => $structure ) {
+				$generated_content_structures[ $home_slug ] = $custom_structure;
+			}
```
Comment (arunshenoy99) [inline includes/SiteGen/SiteGen.php:451]: Do we need to use AI to generate content structures if we are going to replace them anyway with the hardcoded array?

File: includes/SiteGen/SiteGen.php [hunk 2]
```diff
@@ -438,9 +439,21 @@ public static function get_home_pages( $site_description, $content_style, $targe
...
+			$generated_patterns['hero-custom'] = array_pad( array(), 3, Patterns::get_custom_pattern() );
```
Comment (arunshenoy99) [inline includes/SiteGen/SiteGen.php:452]:
```suggestion
			$generated_patterns['hero-custom'] = array_pad( array(), 3, Patterns::get_custom_hero_pattern() );
```

File: includes/SiteGen/SiteGen.php [hunk 3]
```diff
@@ -438,9 +439,21 @@ public static function get_home_pages( $site_description, $content_style, $targe
 			self::cache_sitegen_response( 'homepages', $generated_homepages );
 		}
 
+		$generated_patterns = self::get_sitegen_from_cache( 'generatedPatterns' );
+
+		// check if custom hero patterns needs to be added
+		$site_classification = self::get_sitegen_from_cache( 'siteclassification' );
+		if ( Patterns::check_custom_hero_needed( $site_classification ) ) {
```
Comment (arunshenoy99) [inline includes/SiteGen/SiteGen.php:446]:
```suggestion
		if ( Patterns::check_custom_content_structure_needed( $site_classification ) ) {
```

Comment (abhijitb) [reply includes/SiteGen/SiteGen.php]: done!

File: includes/SiteGen/SiteGen.php (hunk 2 above)
Comment (abhijitb) [reply includes/SiteGen/SiteGen.php]: done!

File: includes/SiteGen/SiteGen.php (hunk 1 above)
Comment (abhijitb) [reply includes/SiteGen/SiteGen.php:451]: As discussed, the logic to make a call to AI for generating homepages right now depends on whether the content structures are saved in cache or not. We cannot change or hardcode the content structure before that has happened as the response contains the generated patterns as well, which are used to construct the homepages.

File: includes/Patterns.php [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+				'trades-repair-services' => false,
+				'hr-recruiting'          => false,
+				'legal'                  => false,
+				'marketing'              => false,
+				'pr-communications'      => false,
+				'real-estate-management' => false,
+				'outdoors'               => false,
+				'shopping-retail'        => false,
+				'weddings'               => false,
+				'other'                  => false,
```
[the remark this answers is not in the document]
Comment (abhijitb) [reply includes/Patterns.php:60]: Good point. For now, we can leave this here and take up moving of this data in a separate Jira ticket. https://jira.newfold.com/browse/PRESS0-1685

**Review by arunshenoy99 — APPROVED** [review]

**Pull request merged**